src/cryptography/hazmat/_types.py

```diff
@@ -27,3 +27,10 @@
     dsa.DSAPrivateKey,
     ec.EllipticCurvePrivateKey,
 ]
+_PRIVATE_KEY_TYPES_WITH_SERIALIZATION = typing.Union[
```

Comment:
Should we consider just making PrivateKey and PrivateKeyWithSerialization the same thing? If we ever support opaque key references we can just make the private_bytes method raise an exception, and it would significantly simplify this typing challenge.

Reply:
Yeah, I'm pro merging them. Separate PR for that though, I suppose.